### PR TITLE
6.2wip

### PR DESCRIPTION
**Pull request by jeffreytoffoli**
Updates for v6.2

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -20,20 +20,32 @@
 ## API Guide
 
 * [API overview](api-guide/api-overview/README.md)
+  * [API Clients](api-guide/api-overview/api-clients.md)
```
Comment (zeiler) [inline SUMMARY.md:23]: something seems weird with this diff as API Clients are already in this location in master.

Comment (zeiler) [reply SUMMARY.md]: i merged in master and resolved conflicts. please double check these files: 
```
(v1) zeiler@matt-zeiler-mbp:~/work/ours/docs$ git checkout 6.2wip
Branch '6.2wip' set up to track remote branch '6.2wip' from 'origin'.
Switched to a new branch '6.2wip'
(v1) zeiler@matt-zeiler-mbp:~/work/ours/docs$ git merge master
Auto-merging portal-guide/concept/README.md
CONFLICT (add/add): Merge conflict in portal-guide/concept/README.md
Auto-merging api-guide/workflows/workflow-predict.md
Auto-merging api-guide/inputs/create-get-update-delete.md
Auto-merging api-guide/concept/languages.md
CONFLICT (add/add): Merge conflict in api-guide/concept/languages.md
Auto-merging api-guide/concept/concepts.md
CONFLICT (add/add): Merge conflict in api-guide/concept/concepts.md
Auto-merging SUMMARY.md
CONFLICT (content): Merge conflict in SUMMARY.md
Auto-merging CHANGELOG.md
CONFLICT (content): Merge conflict in CHANGELOG.md
Automatic merge failed; fix conflicts and then commit the result.
```

File: SUMMARY.md [hunk 2]
```diff
@@ -40,6 +43,9 @@
   * [Evaluate](portal-guide/model/evaluate/README.md)
     * [Interpreting evaluations](portal-guide/model/evaluate/interpreting-evaluations.md)
 * [Workflows](api-guide/workflows.md)
+  * [Create, get, update, delete](api-guide/workflows/create-get-update-delete.md)
```
Comment (zeiler) [inline SUMMARY.md:46]: make uppercase like other sections.

Comment (zeiler) [reply SUMMARY.md:46]: same below, i think all titles should be made consistent.

File: api-guide/concept/knowledge_graph.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
+The Clarifai knowledge graph enables you to map the custom concepts in your model to a common set of concepts. The knowledge graph makes it possible to link data across multiple custom and pre-built models in a meaningful way.
+
+The knowledge graph is organized by the relationship between concepts as either hypernyms, hyponyms, and synonyms.
+
+*Hypernym* is a word with a broader meaning that encapsulates words with more specific meaning falling under it. Cutlery will be a hypernym to knives.
```
Comment (zeiler) [inline api-guide/concept/knowledge_graph.md:5]: for hyper and hypo we shold use the examples from the proto docs like this:
```
  // 1) For example, 'hyponym' is a type of predicate which represents 'is_a_kind_of' relation so
  // the following relationship:
  // 'honey' (subject), 'hyponym' (predicate), 'food' (object)
  // Can more easily be read as:
  // 'honey' 'is a kind of' 'food'
  //
  //
  // 2) The 'hypernym' relation is the opposite of 'hyponym' and when you add one of the
  // relationships the opposite will automatically appear for you in queries.
  //
  // The 'hypernym' can be read as 'is a parent of' so:
  // 'food' (subject), 'hypernym' (predicate), 'honey' (object)
  // Can more easily be read as:
  // 'food' is a parent of 'honey'
  //
  // 3) The 'synonym' relation defines two concepts that essential mean the same thing. This
  // is more like a "is" relationship. So for example a 'synonym' relationship could be:
  // "puppy" is "pup"
  // The reverse is also true once the former is added so:
  // "pup" is "puppy"
  // will appear in queries as well.
```
the "is a" is the only way to make sense of hyponyms in my opinion. it's also important to talk about the subject and object when defining these because the opposite direction is also possible in the relationship.

File: api-guide/inputs/collectors.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+Collectors capture input data for your app. They enable you to pipe in data from production models automatically, and are the key to unlocking many platform training capabilities like active learning. Collectors are available with Essential and Enterprise plans to help you manage data ingestion at scale.
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:1]: @alfredoclarifai we should have icons for essential and enterprise so everywhere we mention features tied to those plans we use the icons and start building that in people's mindsets.

File: api-guide/concept/knowledge_graph.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+*Hyponym* is a word with more specific meaning and is the opposite of hypernym. Knives would be a hyponym to cutlery.
+*Synonym* is a word with similar meaning to another word.
+
+You can create these mappings in your application with a few easy steps.
```
Comment (zeiler) [inline api-guide/concept/knowledge_graph.md:9]: how come we're showing portal in the API guide? If I'm reading the API guide I want to use the API and need to see the calls.

File: api-guide/inputs/collectors.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+You can create app-level collectors to monitor specific models and specify sampling rules for triggering data ingestion.
+
+
+## Creating a new collector
+
+Collectors help you to feed your models with real-world training data. This data can be taken from models that you have already deployed to production. Just create a collector within your app and set it up to ingest data from another model when new inputs are "posted" to this model.
+
+![](../../images/create_new_collector.jpg)
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:10]: i'm in the api guide so I want to see the API calls that make collectors possible. they have full CRUD

File: api-guide/inputs/collectors.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+### Collector ID
+Give your collector a useful and descriptive name.
+
+### Description
+Provide additional details.
+
+
+### Pre-queue workflow
+
+Pre-queue workflows allow you to specify sampling rules for triggering data ingestion.
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:21]: i wouldn't say "sampling" because that's only one type of thing you can do with a workflow of models. More generally I would say "filtering".

File: api-guide/inputs/collectors.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+
+In many scenarios, you will only want to ingest a sample, or subset of a given data source into your app. Pre-queue workflows allow you to pre-process your inputs so that you can sample and filter your new data before it is ever added to your app.
+
+Common pre-queue workflows are designed to:
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:25]: we don't provide any common workflows so I wouldn't say they are designed to

Comment (zeiler) [reply api-guide/inputs/collectors.md:25]: common use case for pre-queue workflows are:

File: api-guide/inputs/collectors.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+
+* Randomly sample inputs
+* Filter inputs by metadata
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:28]: we don't have this metadata one (yet).

File: api-guide/inputs/collectors.md [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+
+
+### Post Inputs key
+
+Select the API key that you would like to use to allow new inputs to be posted to your app. This API key must have the PostInputs scope, since it grants the collector the authority to POST inputs to your app.
+
+
+### Caller
+
+Choose to only ingest inputs from a specified user, or anyone who posts new inputs to a model.  The user is referred to as the "caller" since they are making API calls as they post new inputs to their model. Just input the caller's user ID, or select "Any caller" to accept inputs from anyone posting new inputs to the model.
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:42]: i don't think the app owner can select the option to collect from any user so should remove that, they can only collect from themselves i believe? should try it out with external email account.

File: api-guide/inputs/collectors.md [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+
+### Source
+
+Select the model that you would like to collect from, and the collector will automatically post the new inputs to your app. Simply enter your model name, or model ID number. You can select the model that you would like to collect from in the drop down menu.
```
Comment (zeiler) [inline api-guide/inputs/collectors.md:47]: we never accept model names in requests, remove that. always IDs everywhere.

File: api-guide/workflows/README.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
+# Workflows
+
+Workflows are a graph of computation that encompass one or more Clarifai or Custom model\(s\). Every workflow is attached to one of your applications. Under each workflow, you will see a list of the Clarifai Models and all custom models in that application when selecting models to add to your workflow. With Workflow Predict, you will be able to reduce the latency and in turn make your product more efficient.
```
Comment (zeiler) [inline api-guide/workflows/README.md:3]: i wouldn't talk about latency that much, it's more about running your complex business logic with us.

File: api-guide/workflows/README.md [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+
+Note: this won't have any impact on the price you are charged per call. You will still be charged for the same operation if it were separate calls to the API.
```
Comment (zeiler) [inline api-guide/workflows/README.md:5]: make this more clear. if you do a predict with a workflow you're charged for the prediction of each model in the workflow over the data that is input to the workflow as if they were separate calls.

File: api-guide/workflows/README.md [hunk 14]
```diff
@@ -0,0 +1,19 @@
...
+
+## Workflow Setup
+
+To set up a workflow, you will need to head over to the [Applications page](https://portal.clarifai.com/apps) through your account. From there, you will need to select which application you want to create the workflow under.
+
+![Image showing the top-level Applications page on the Clarifai Developer website](../images/application-screen-new.png)
```
Comment (zeiler) [inline api-guide/workflows/README.md:11]: again i'm in API guide, i want to see API calls.

File: api-guide/workflows/README.md [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+Then under that application, you will see a section labeled "App Workflows" and a button to "Create Workflow".
+
+![Image showing My First Application and the Create Workflow button underneath the Create API Key](../images/create-workflow-new.png)
```
Comment (zeiler) [inline api-guide/workflows/README.md:15]: this create workflow shot is very old, should be updated to the one in model mode which will be the future.

File: api-guide/workflows/README.md [hunk 16]
```diff
@@ -0,0 +1,19 @@
...
+
+After that, the page will reveal a new workflow form to fill out. Fill out the Workflow ID field, this will be used to make the API call, so make sure to give it something URL friendly! Included there, you will also a list that consists of a model field and a version associated with it. For Clarifai Models, you will be mandated to use the latest version. For your custom models, you will be able to [select the version of your model](https://github.com/Clarifai/docs/tree/5882f46bd17affcd85ed3e2ec98f4d6f355b58a9/models.md#list-model-versions)&lt;/a&gt;. To add another model, you will just click underneath your latest addition on the "Add Model". The max limit of models associated with any given workflow is 5 models. If you would like to remove a model, there is a large X that will allow you to remove a model. Once you have finished adding everything, press the "Save Workflow" button and that will save the state of your workflow. Now you are ready to predict using your brand new workflow. You can edit a given workflow at any time, in case you don't like it.
```
Comment (zeiler) [inline api-guide/workflows/README.md:17]: there is a link to github here rather than docs.clarifai.com

Comment (zeiler) [reply api-guide/workflows/README.md:17]: max limit is not 5, it's much higher now at 20

File: portal-guide/concept/knowledge_graph.md [hunk 17]
```diff
@@ -0,0 +1,33 @@
+The Clarifai Knowledge graph enables you to map the custom concepts in your model to a common set of concepts. The knowledge graph makes it possible to link data across multiple custom and pre-built models in a meaningful way.
+
+The knowledge graph is organized by the relationship between concepts as either hypernyms, hyponyms, and synonyms.
+
+*Hypernym* is a word with a broader meaning that encapsulates words with more specific meaning falling under it. Cutlery will be a hypernym to knives.
```
Comment (zeiler) [inline portal-guide/concept/knowledge_graph.md:5]: is this an exact copy of the other API guide file?

File: api-guide/workflows/input_nodes.md [hunk 18]
```diff
@@ -0,0 +1,56 @@
+The outputs from one workflow can be used as the inputs in another workflow. This allows you to link together the models in a graph. This ability to link together models is the key to building targeted AI solutions.
+
+## Supported input and output types
```
Comment (zeiler) [inline api-guide/workflows/input_nodes.md:3]: i think this entire section will be forever out of date and hard to keep in sync. Why don't we just point them at model mode to see the variety of input and output types and use this space to talk about some example workflow graphs that require input nodes like auto annotation workflows or anything with branching really.

File: api-guide/inputs/collectors.md (hunk 4 above)
Comment (jeffreytoffoli) [reply api-guide/inputs/collectors.md]: still waiting on these icons, moving to "general docs improvements"

File: api-guide/tutorials/auto-annotation.md [hunk 19]
```diff
@@ -0,0 +1,1847 @@
...
+                    }
+                }
+            }
+        }
+    }'
+```
+{% endtab %}
+{% endtabs %}
+
+### Create a "write success as me" model
```
Comment (zeiler) [inline api-guide/tutorials/auto-annotation.md:783]: Create a "write success as me" Annotation Writer model

File: api-guide/tutorials/auto-annotation.md [hunk 20]
```diff
@@ -0,0 +1,1847 @@
...
+                    }
+                }
+            }
+        }
+    }'
+```
+{% endtab %}
+{% endtabs %}
+
+### Create a "write pending as me" model
```
Comment (zeiler) [inline api-guide/tutorials/auto-annotation.md:929]: Create a "write pending as me" Annotation Writer model

File: api-guide/tutorials/auto-annotation.md [hunk 21]
```diff
@@ -0,0 +1,1847 @@
...
+{% endtabs %}
+
+### Create the workflow
+
+We will now joint all the models together into a single workflow.
+Every input will be predicted by general embed model to generate embedding. The output of the embed model (embeddings) will be sent to general concept to predict concept and cluster model. Then the concept model's output (a list of concepts) will be sent to concept mapper model which maps Clarifai concept to your concept, `people`, `man` and `adult` in this case. Then the mapped concepts will be sent to both concept thresholds models (`GREATER THAN` and `LESS THAN`). `GREATER THAN` model will filter out the concept if it lower than corresponding value you defined in model and send the final concept list to `write success as me` model which labels the input with these concepts (your app concepts only) as you with `success` stauts. You can train or search on these concepts immediately. At the mean time, `LESS THAN` model will filter out the concept if it higher than ccorresponding valud you defined in model and send the final concept list to `write pending as me` model which labels the input with these concepts (your app concepts only) as you with `pending` status.
```
Comment (zeiler) [inline api-guide/tutorials/auto-annotation.md:1045]: valud -> value

File: api-guide/tutorials/auto-annotation.md [hunk 22]
```diff
@@ -0,0 +1,1847 @@
...
+            }
+        ]
+    }'
+```
+{% endtab %}
+{% endtabs %}
+
+### Make the new workflow app's default
+
+Make this the default workflow in the app, so it will every time we add an input.
```
Comment (zeiler) [inline api-guide/tutorials/auto-annotation.md:1528]: Make this the default workflow in the app, so it will run every time we add an input, and therefore execute the auto annotation process for us!

File: api-guide/workflows/input_nodes.md [hunk 23]
```diff
@@ -0,0 +1,514 @@
...
+
+![](../../images/input_nodes.jpg)
+
+## Supported input and output types
+
+To view your available models, just open your app and click Model Mode icon on the left hand side of the screen. From here just click the Create a Custom Model button in the top righthand corner of the screen.
+
+Different models accept different types of inputs and return different types of outputs. Some examples include:
+
+#### Inputs
```
Comment (zeiler) [inline api-guide/workflows/input_nodes.md:11]: the names of these inputs/output have changed. See dev model mode for the actual types. 

And make a comment that they are named after the fields in the Data object of our API. This object is used in inputs, annotations, models and workflows. There could really be a docs section for Data object in future as it's pretty crucial data container used throughout our platform. it's how we can pass outputs from one model as inputs to next model in a workflow.

File: portal-guide/concept/knowledge_graph.md [hunk 24]
```diff
@@ -0,0 +1,36 @@
+The Clarifai Knowledge Graph lets you map your custom concepts to a common set of concepts understood by all applications on the Clarifai platform. The knowledge graph makes it possible to link data across multiple custom and pre-built models in a meaningful way.
```
Comment (zeiler) [inline portal-guide/concept/knowledge_graph.md:1]: is there a title needed on this page?

File: portal-guide/concept/knowledge_graph.md [hunk 25]
```diff
@@ -0,0 +1,36 @@
...
+
+The Knowledge Graph uses Clarifai's concept mapping model to establish a hierarchical relationship between your concepts. and to uses three different *predicates* to organize your concepts: hypernyms, hyponyms, and synonyms.
+
+**Hyponym** represents an 'is a kind of' relation. The following relationship: 'honey' (subject), 'hyponym' (predicate), 'food' (object) is more easily be read as 'honey' 'is a kind of' 'food'.
+
+**Hypernym** is the opposite of 'hyponym'. When you add one of the relationships the opposite will automatically appear for you in queries. The 'hypernym' can be read as 'is a parent of' so: 'food' (subject), 'hypernym' (predicate), 'honey' (object)can more easily be read as:'food' is a parent of 'honey'.
```
Comment (zeiler) [inline portal-guide/concept/knowledge_graph.md:7]: `(object)can` -> `(object) can`

File: api-guide/concept/knowledge_graph.md [hunk 26]
```diff
@@ -0,0 +1,362 @@
+The Clarifai Knowledge Graph lets you map your custom concepts to a common set of concepts understood by all applications on the Clarifai platform. The knowledge graph makes it possible to link data across multiple custom and pre-built models in a meaningful way.
+
+The Knowledge Graph uses Clarifai's concept mapping model to establish a hierarchical relationship between your concepts. and to uses three different *predicates* to organize your concepts: hypernyms, hyponyms, and synonyms.
+
+**Hyponym** represents an 'is a kind of' relation. The following relationship: 'honey' (subject), 'hyponym' (predicate), 'food' (object) is more easily be read as 'honey' 'is a kind of' 'food'.
+
+**Hypernym** is the opposite of 'hyponym'. When you add one of the relationships the opposite will automatically appear for you in queries. The 'hypernym' can be read as 'is a parent of' so: 'food' (subject), 'hypernym' (predicate), 'honey' (object)can more easily be read as:'food' is a parent of 'honey'.
```
Comment (zeiler) [inline api-guide/concept/knowledge_graph.md:7]: `(object)can` > `(object) can`

File: portal-guide/model/deep-training.md [hunk 27]
```diff
@@ -0,0 +1,87 @@
...
+* **A custom tailored dataset**
+* **Accurate labels**
+* **Expertise and time to fine tune models**
+
+
+## Template types
+
+You can take advantage of a variety of templates when building your deep trained models. Templates give you the control to choose the specific architecture used by your neural network, and also define a set of hyperparameters that you can use to fine-tune the way that your model learns.
+
+#### Classification
```
Comment (zeiler) [inline portal-guide/model/deep-training.md:17]: same comments as I made in the API section.
